internal/clusterfeature/helm.go

```diff
@@ -0,0 +1,183 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusterfeature
+
+import (
+	"context"
+
+	"github.com/banzaicloud/pipeline/helm"
+	"github.com/goph/emperror"
+	"github.com/goph/logur"
+	k8sHelm "k8s.io/helm/pkg/helm"
+	"k8s.io/helm/pkg/proto/hapi/release"
+)
+
+// HelmService interface for helm operations
+type HelmService interface {
+	// InstallDeployment installs a feature to the given cluster
+	InstallDeployment(ctx context.Context, orgName string, kubeConfig []byte, namespace string,
+		deploymentName string, releaseName string, values []byte, chartVersion string, wait bool) error
+
+	// DeleteDeployment removes a feature to the given cluster
+	DeleteDeployment(ctx context.Context, kubeConfig []byte, releaseName string) error
+
+	UpdateDeployment(ctx context.Context, orgName string, kubeConfig []byte, namespace string,
+		deploymentName string, releaseName string, values []byte, chartVersion string) error
+}
+
+// component in chrge for installing features from helm charts
+type featureHelmService struct {
+	logger logur.Logger
+}
+
+func (hs *featureHelmService) InstallDeployment(
+	ctx context.Context,
+	orgName string,
```

Comment:
I would extract `orgName`, `kubeConfig`, `namespace`, `deploymentName`, `releaseName`, `values` and `chartVersion` to a struct (e.g. `DeploymentParams`) and use it as a single parameter in `InstallDeployment` and `UpdateDeployment` to reduce the number of parameters of these methods.

Reply:
i plan this as part of the helm service abstraction across the pipeline project (separate tasks ...)